src/cloud-element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -1021,6 +1023,105 @@ export default class ChangeElementTemplateHandler {
+   * can only exist in `bpmn:ExtensionElements`.
+   *
+   * @param {djs.model.Base} element
+   * @param {Object} oldTemplate
+   * @param {Object} newTemplate
+   */
+  _updateScriptTask(element, oldTemplate, newTemplate) {
+    const bpmnFactory = this._bpmnFactory,
+          commandStack = this._commandStack;
+
+    const newProperties = newTemplate.properties.filter((newProperty) => {
+      const newBinding = newProperty.binding,
+            newBindingType = newBinding.type;
+
+      return newBindingType === ZEEBE_SCRIPT_TASK;
+    });
+
+    const businessObject = this._getOrCreateExtensionElements(element);
+    let scriptTask = findExtension(businessObject, 'zeebe:Script');
+
+    // (1) remove old called element if no new properties specified
+    if (!newProperties.length) {
+      commandStack.execute('element.updateModdleProperties', {
+        element,
+        moddleElement: businessObject,
+        properties: {
+          values: without(businessObject.get('values'), scriptTask)
+        }
+      });
+
+      return;
+    }
+
+    newProperties.forEach((newProperty) => {
+      const oldProperty = findOldProperty(oldTemplate, newProperty),
+            newPropertyValue = getDefaultValue(newProperty),
+            propertyName = newProperty.binding.property;
+
+      // (2) update old called element
+      if (scriptTask) {
+
+        if (!shouldKeepValue(scriptTask, oldProperty, newProperty)) {
+          const properties = {
+            [propertyName]: newPropertyValue
+          };
+
+          commandStack.execute('element.updateModdleProperties', {
+            element,
+            moddleElement: scriptTask,
+            properties
+          });
+        }
+      }
+
+      // (3) add new called element
+      else {
+        const properties = {
+          [propertyName]: newPropertyValue
+        };
+
+        scriptTask = createScriptTask(properties, bpmnFactory);
+
+        scriptTask.$parent = businessObject;
+
+        commandStack.execute('element.updateModdleProperties', {
+          element,
+          moddleElement: businessObject,
+          properties: {
+            values: [ ...businessObject.get('values'), scriptTask ]
+          }
+        });
+      }
+    });
+
+    // (4) remove properties no longer templated
+    const oldProperties = oldTemplate && oldTemplate.properties.filter((oldProperty) => {
+      const oldBinding = oldProperty.binding,
+            oldBindingType = oldBinding.type;
+
+      return oldBindingType === ZEEBE_SCRIPT_TASK && !newProperties.find(
+        (newProperty) => newProperty.binding.property === oldProperty.binding.property
+      );
+    }) || [];
+
+    oldProperties.forEach((oldProperty) => {
+      const properties = {
+        [oldProperty.binding.property]: undefined
+      };
+
+      commandStack.execute('element.updateModdleProperties', {
+        element,
+        moddleElement: scriptTask,
+        properties
+      });
+    });
```

Comment:
I guess this is not really needed for a script task that has 2 mandatory properties only... but also, not entirely sure.